test/fs/test_fs_readdir_ino_matches_stat_ino.c

```diff
@@ -0,0 +1,60 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <assert.h>
+#include <fcntl.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <assert.h>
+#include <errno.h>
+#include <string.h>
+#include <dirent.h>
+
+// TODO: Consider merging this with test/dirent/test_readdir.c?
+// That test doesn't work on nodefs, which has to be fixed before these two
+// tests can be merged .
+
+void setup() {
+  // If the node working directory does not have execute permissions, then
+  // `fs.readdir` will fail. For that reason we have to work in a subdirectory
+  // and remove execute permissions from that.
```

Comment:
I don't understand.. `fs.readdir` should fail on any directory that doesn't have the execute bit set, regardless of weather that is CWD or a sub directory, right?

Reply:
Apparently not.

Reply:
I suppose I can drop the chmod, but cding into a subdirectory does fix the test, at least when I run it locally. I don't really have a clear theory as to why since it seems a bit illogical.

Reply:
Ah, yes, I misremembered.  The `x` bit is about traversing the heirachy, not list-ability.  You can list directory than don't have the `x` bit set.  That is what the `r` bit is for.